hq_superset/commands.py

```diff
@@ -0,0 +1,55 @@
+import requests
+import json
+
+
+def subscribe_data_sources(superset_base_url, hq_base_url, data_sources_file_path, username, apikey):
+    from hq_superset.services import (
+        _get_or_create_oauth2client,
+        datasource_subscribe,
+    )
+
+    def data_sources_by_domain():
+        with open(data_sources_file_path) as file:
+            datasource_ids_by_domain = json.loads(file.read())
+
+        # Validate file content
+        assert isinstance(datasource_ids_by_domain, dict)
+        for domain, ds_ids in datasource_ids_by_domain.items():
+            if not isinstance(ds_ids, list):
+                raise Exception(f"{domain} expected a list of data source IDs")
+        return datasource_ids_by_domain
+
+    webhook_url = f"{superset_base_url}/change/"
+    token_url = f"{superset_base_url}/token"
+
+    failed_data_sources_by_id = {}
+    for domain, datasource_ids in data_sources_by_domain().items():
+        print(f"Handling domain: {domain}")
+        for datasource_id in datasource_ids:
+            print(f"Syncing datasource: {datasource_id}")
+            client = _get_or_create_oauth2client(domain)
+
+            endpoint = datasource_subscribe(domain, datasource_id)
+            data = {
+                'webhook_url': webhook_url,
+                'token_url': token_url,
+                'client_id': client.client_id,
+                'client_secret': client.get_client_secret(),
+            }
+            response = requests.post(
+                f"{hq_base_url}/{endpoint}",
```

Comment:
Use `urllib.parse.urljoin` to handle forward slashes gracefully. Django can figure out extra forward slashes, but proxies can make things unpredictable. Safer just to use `urljoin()`.

```suggestion
                urljoin(hq_base_url, endpoint),
```